### DISCUSSION TITLE
undirected many-to-many relationship with one table

### DISCUSSION THREAD

**Discussion by IsHaltEchtSo** [sqlalchemy/sqlalchemy#8329] [Usage Questions] [answered]
Is it possible to build an undirected, many-to-many relationship with only one table and an association table? Like a UserModel where two User-objects can be friends; if User A is a friend of User B, User B has a friend in User A as well. I can only do it directed - like follower/following - but not undirected as of now... for more info, here's the full issue:
[stackoverflow link](https://stackoverflow.com/questions/73099995/undirected-many-to-many-relationship-with-one-table)

**Comment by zzzeek** [comment, marked as answer]
short answer is not really, at least with a single row in the association table.   This came up [here](https://github.com/sqlalchemy/sqlalchemy/issues/6137#issuecomment-808412223) where someone else got part of this working using a bit of a hack and I did some thinking about it - but it involves two rows in the association table in order to represent the two directions, you might want to play with that.

**Reply by IsHaltEchtSo** [reply]
Thanks Mike, I'll play around with it and see how it works! :)
